src/PKSim.Core/Services/RangeChartsDataCreator.cs

```diff
@@ -67,18 +67,23 @@ private void connectCurvesDefinedIn(ChartData<RangeXValue, RangeYValue> chart)
 
       private void connectCurvesDefinedIn(PaneData<RangeXValue, RangeYValue> panes)
       {
-         var orderedCurveByXvalue = panes.Curves.Where(x => x.XValues.Any())
+         var orderedCurveByXValue = panes.Curves.Where(x => x.XValues.Any())
             .OrderBy(x => x.XValues[0]).ToList();
 
          //Some curves are invalid, we do not try to connect anything in that case
-         if (orderedCurveByXvalue.Count != panes.Curves.Count)
+         if (orderedCurveByXValue.Count != panes.Curves.Count)
             return;
 
-         for (int i = 1; i < orderedCurveByXvalue.Count; i++)
+         for (int i = 1; i < orderedCurveByXValue.Count; i++)
          {
-            var firstCurve = orderedCurveByXvalue[i - 1];
-            var secondCurve = orderedCurveByXvalue[i];
-            firstCurve.Add(secondCurve.XValues[0], secondCurve.YValues[0]);
+            var firstCurve = orderedCurveByXValue[i - 1];
+            var secondCurve = orderedCurveByXValue[i];
+            // Only connect points if curves already belong to one and only that were split for some binning reasons. 
+            // e.g if the current X is after the next X, those curves clearly do not belong together
+            if (firstCurve.XValues.Last().X < secondCurve.XValues[0].X)
+            {
+              firstCurve.Add(secondCurve.XValues[0], secondCurve.YValues[0]);
```

Comment:
This is the change to fix the issue with the range completely messed up. In this case, if the last point of the first curve is before the first point of the second curve, we bring them together. I am not sure how to do it in a better way since there is no way to say whether curve are related or not (e.g. grouping by sim name or gender creates the same structure, but clearly only  one makes sense to connect)